packages/voting-contract/src/lib.rs

```diff
@@ -405,18 +376,22 @@ pub fn list_votes_by_voter<Q: CustomQuery>(
     start_after: Option<u64>,
     limit: usize,
 ) -> StdResult<VoteListResponse> {
-    let start = start_after.map(Bound::exclusive);
     let voter_addr = deps.api.addr_validate(&voter)?;
-
-    let votes: StdResult<Vec<_>> = BALLOTS_BY_VOTER
-        .prefix(&voter_addr)
+    // PrimaryKey of that IndexMap is (proposal_id, voter_address) -> (u64, Addr)
```

Comment:
Also, pk is (the last ) part of a (multi-)index key. That's why after fixing the voter addr with `prefix()` you still to range over the rest of the index key (which matches / corresponds to the pk).